api/src/main/java/com/github/retrooper/packetevents/manager/InternalPacketListener.java

```diff
@@ -88,7 +88,7 @@ else if (event.getPacketType() == PacketType.Configuration.Server.REGISTRY_DATA)
 
             if (packet.getElements() != null) { // 1.20.2 to 1.20.5
                 SynchronizedRegistriesHandler.handleRegistry(user, packet.getServerVersion().toClientVersion(),
-                        packet.getRegistryKey(), packet.getElements());
+                        packet.getRegistryKey(), packet.getElements(), !PacketEvents.getAPI().getInjector().isProxy());
```

Comment:
Here we check if we are running on proxy.
If not, we cache the registries globally to save memory.